tasks/install_targz.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name: downloading bundled agent file
+  get_url:
+    url: "{{ nrinfragent_custom_targz }}"
+    dest: "{{ nrinfragent_custom_tmpdir }}/newrelic-infra.tar.gz"
+    force: yes
+- name: uncompressing bundled agent file
+  unarchive:
```

Comment:
We should add a `creates` here so that if a tar has already been uncompressed, the task is skipped. 

You can look for the `creates` option here  https://docs.ansible.com/ansible/latest/modules/unarchive_module.html?highlight=unarchive

I think for this it would be better to have an additional directory like:

`/opt/newrelic_infra/linux_${tarball_version}_${arch}`

That way you can put a `creates:  "/opt/newrelic_infra/linux_${tarball_version}_${arch}"/newrelic-infra`, with this the untar for a given `version/arch` would only be executed once. This task should notify the `running installer`, because we also want to run the installer only once.

Reply:
Same as before, I don't see the scenario where it has to be skipped because during the agent installation this file is not present, and during the agent update process, this file needs to be re-created/overwriten.

Reply:
Just let me point that `/tmp` and the `unarchive` destination are folders of the target machine, not from the Ansible controller (In that case it would have all the sense).

Reply:
yeah, but on a normal run, when the agent is already uncompress, why would you want to execute the action again? In case you leave it like this, make sure that if the installer is executed on every ansible run, it doesn't trigger a service restart.